src/core_read.cpp

```diff
@@ -21,13 +21,16 @@
 #include <string>
 
 namespace {
-
-opcodetype ParseOpCode(const std::string& s)
+class OpCodeParser
 {
-    static std::map<std::string, opcodetype> mapOpNames;
+private:
+    std::map<std::string, opcodetype> mapOpNames;
 
-    if (mapOpNames.empty()) {
-        for (unsigned int op = 0; op <= MAX_OPCODE; op++) {
+public:
+    OpCodeParser()
+    {
+        for (unsigned int op = 0; op <= MAX_OPCODE; ++op)
+        {
```

Comment:
Nit: coding style. Opening brace for anything but classes/functions is on the same line.